Source/Scene/PointPrimitiveCollection.js

```diff
@@ -0,0 +1,976 @@
+        DeveloperError,
+        EncodedCartesian3,
+        IndexDatatype,
+        CesiumMath,
+        Matrix4,
+        PrimitiveType,
+        BufferUsage,
+        DrawCommand,
+        ShaderSource,
+        VertexArrayFacade,
+        PointPrimitiveCollectionFS,
+        PointPrimitiveCollectionVS,
+        PointPrimitive,
+        BlendingState,
+        Pass,
+        SceneMode) {
+    "use strict";
+
+    var SHOW_INDEX = PointPrimitive.SHOW_INDEX;
+    var POSITION_INDEX = PointPrimitive.POSITION_INDEX;
+    var COLOR_INDEX = PointPrimitive.COLOR_INDEX;
+    var OUTLINE_COLOR_INDEX = PointPrimitive.OUTLINE_COLOR_INDEX;
+    var OUTLINE_WIDTH_INDEX = PointPrimitive.OUTLINE_WIDTH_INDEX;
+    var PIXEL_SIZE_INDEX = PointPrimitive.PIXEL_SIZE_INDEX;
+    var SCALE_BY_DISTANCE_INDEX = PointPrimitive.SCALE_BY_DISTANCE_INDEX;
+    var TRANSLUCENCY_BY_DISTANCE_INDEX = PointPrimitive.TRANSLUCENCY_BY_DISTANCE_INDEX;
+    var NUMBER_OF_PROPERTIES = PointPrimitive.NUMBER_OF_PROPERTIES;
+
+    var attributeLocations = {
+        positionHighAndSize : 0,
+        positionLowAndOutline : 1,
+        compressedAttribute0 : 2,        // color, outlineColor, pick color
+        compressedAttribute1 : 3,        // show, translucency by distance, some free space
+        scaleByDistance : 4
+    };
+
+    /**
+     * A renderable collection of glPoints.
+     * <br /><br />
+     * PointPrimitives are added and removed from the collection using {@link PointPrimitiveCollection#add}
+     * and {@link PointPrimitiveCollection#remove}.  PointPrimitives in a collection automatically share textures
+     * for images with the same identifier.
+     *
+     * @alias PointPrimitiveCollection
+     * @constructor
+     *
+     * @param {Object} [options] Object with the following properties:
+     * @param {Matrix4} [options.modelMatrix=Matrix4.IDENTITY] The 4x4 transformation matrix that transforms each pointPrimitive from model to world coordinates.
+     * @param {Boolean} [options.debugShowBoundingVolume=false] For debugging only. Determines if this primitive's commands' bounding spheres are shown.
+     *
+     * @performance For best performance, prefer a few collections, each with many pointPrimitives, to
+     * many collections with only a few pointPrimitives each.  Organize collections so that pointPrimitives
+     * with the same update frequency are in the same collection, i.e., pointPrimitives that do not
+     * change should be in one collection; pointPrimitives that change every frame should be in another
+     * collection; and so on.
+     *
+     * @see PointPrimitiveCollection#add
+     * @see PointPrimitiveCollection#remove
+     * @see PointPrimitive
+     * @see LabelCollection
+     *
+     * @example
+     * // Create a pointPrimitive collection with two pointPrimitives
+     * var pointPrimitives = new Cesium.PointPrimitiveCollection();
+     * pointPrimitives.add({
+     *   position : { x : 1.0, y : 2.0, z : 3.0 },
```

Comment:
This is a really old example...use `Cartesian3` here and throughout.  Also update `BillboardCollection` and `LabelCollection` please.